Fix regressions in link UI

jasmussen opened the pull request:
Recently we merged in some improvements to the design of input fields. As part of that, some regressions to the link UI on buttons and link modals were introduced. This PR fixes those.

Screenshots:

<img width="520" alt="screen shot 2018-04-17 at 09 51 45" src="https://user-images.githubusercontent.com/1204802/38857774-5575d3f6-422a-11e8-978e-44d862febb0e.png">

<img width="684" alt="screen shot 2018-04-17 at 10 27 49" src="https://user-images.githubusercontent.com/1204802/38857776-56abf232-422a-11e8-92c9-152cb514cd56.png">

<img width="697" alt="screen shot 2018-04-17 at 10 27 56" src="https://user-images.githubusercontent.com/1204802/38857778-5803f666-422a-11e8-83cc-217de70305af.png">

<img width="321" alt="screen shot 2018-04-17 at 10 28 03" src="https://user-images.githubusercontent.com/1204802/38857779-5978b0ea-422a-11e8-824e-ac6a7ab82147.png">

As part of this, I did some major cleanup too, to use more SCSS variables stored in the variables stylesheet. I also added some admin scheme color to the selected item in the suggest dropdown. In fact, quite a bit of CSS was messy, duplicate and redundant, so this is quite a nice cleanup to get in. But it also means that it's a fair bit of deep surgery, and would be good to test. 

@karmatosed I know you've filed a few tickets around link interface improvements. Does this PR address some of those?

Commit:
Polish link UI.

The link UI was a bit of a mess, this cleans it up a bit, and fixes some visual glitches as part of the recent input style tweaks.

Commit:
Fix Button block URL

Further optimizes and cleans up the CSS.

karmatosed commented:
This is great, thanks for picking up this. The one we had open about https://github.com/WordPress/gutenberg/issues/6107 : unifying on buttons, is still looking like it has the old interface. That is the one I would love to see changed :)

jasmussen commented:
Thanks Tammie, responded to that other ticket.

youknowriad commented on blocks/url-input/style.scss:
```diff
@@ -30,35 +27,43 @@
-	top: 100%;
-	left: 0;
-	right: -64px; // to match the link modal borders
+	z-index: 0;	// place underneath the input field so focus isn't cropped
```
Should we use the z-index helper instead?

Commit:
Remove z-index

Turns out it worked unexpectedly, and the issue is fixed isn't necessarily important to address.

jasmussen commented:
🔥
